### PR TITLE
planner: Move the Selectivity function from the stats package into cardinality package

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #46358

Problem Summary: Move the Selectivity function from the stats package into cardinality package

### What is changed and how it works?

**No logical change, just some code movement.**

Move the Selectivity function from the stats package.

image

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/cardinality/pseudo.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			if !ok {
+				continue
+			}
+			colExists[col.Info.Name.L] = true
+			if mysql.HasUniKeyFlag(col.Info.GetFlag()) {
+				return 1.0 / float64(coll.RealtimeCount)
+			}
+		case ast.GE, ast.GT, ast.LE, ast.LT:
+			minFactor = math.Min(minFactor, 1.0/pseudoLessRate)
+			// FIXME: To resolve the between case.
```
Comment (hawkingrei) [inline planner/cardinality/pseudo.go:59]: This FIXME was created in 2017.

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**